mypy/newsemanal/semanal.py

```diff
@@ -3823,7 +3823,11 @@ def add_builtin_aliases(self, tree: MypyFile) -> None:
                 self.mark_incomplete(name, tree)
             else:
                 # Test fixtures may be missing some builtin classes, which is okay.
-                pass
+                # Also some targets are not available on Python 2, like collections.ChainMap.
+                # Kill the placeholder if there is one.
+                if name in tree.names:
```

Comment:
This is kind of hacky. Why do we generate the placeholder for things that don't exist? For example, could we include version information in the aliases, and skip the alias definition if the target Python version is too old?

Reply:
As discussed elsewhere, I keep this for test fixtures, but keep track of Python versions for real typeshed stubs.